retention/src/test/java/org/apache/falcon/retention/FeedEvictorTest.java

```diff
@@ -462,6 +462,43 @@ public void testFeedBasePathExists() throws Exception {
         }
     }
 
+    @Test
+    public void testEvictionStatsMetaWithNoPattern() throws Exception {
+        try {
+            Configuration conf = cluster.getConf();
+            FileSystem fs = FileSystem.get(conf);
+            fs.delete(new Path("/"), true);
```

Comment:
@bvellanki The feed definition (FeedEntityParser) currently allows data location without patterns. (see below the example). If we need to enforce data location to have patterns, it should be done at the time of submission. I'm reluctant to do that now since it could have backward compatibility issues and should be done via a separate discuss/JIRA.

<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<feed name="testfeed-data-stats-meta-nopattern" description="Raw customer email feed" xmlns="uri:falcon:feed:0.1">
    <tags>externalSystem=USWestEmailServers</tags>
    <groups>churnAnalysisDataPipeline</groups>
    <frequency>hours(1)</frequency>
    <timezone>UTC</timezone>
    <late-arrival cut-off="hours(1)"/>
    <clusters>
        <cluster name="primaryCluster" type="source">
            <validity start="2016-04-30T01:00Z" end="2017-10-30T10:00Z"/>
            <retention limit="days(1)" action="delete"/>
        </cluster>
    </clusters>
    <locations>
        <location type="data"  path="/tmp/falcon-regression/testfeed-stats-meta-nopattern/data/"/>
        <location type="stats" path="/tmp/stats"/>
        <location type="meta"  path="/tmp/meta"/>
    </locations>
    <ACL owner="ambari-qa" group="users" permission="0x755"/>
    <schema location="/none" provider="/none"/>
</feed>